File: rest-service/src/main/java/mb/lib/query/db/SelectParentJobLinks.java
package mb.lib.query.db;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.util.List;

import io.vulpine.lib.query.util.basic.BasicPreparedListReadQuery;
import mb.lib.model.HashID;
import mb.lib.query.model.BlastJobLink;

public class SelectParentJobLinks
{
  private static final String Query = """
    SELECT
      *
    FROM
      userlogins5.multiblast_job_to_jobs
    WHERE
      job_digest = ?
    ORDER BY
      position ASC
    """;

  private final Connection con;
  private final HashID     childJobID;

  public SelectParentJobLinks(Connection con, HashID childJobID) {
    this.con        = con;
    this.childJobID = childJobID;
  }

  public List<BlastJobLink> run() throws Exception {
    return new BasicPreparedListReadQuery<>(Query, con, Util::parseJobLink, this::prep)
      .execute()
      .getValue();
  }

  private void prep(PreparedStatement ps) throws Exception {
    ps.setBytes(1, childJobID.bytes());
  }
}

File: rest-service/src/main/kotlin/mb/lib/query/db/SelectParentJobLinks.kt
package mb.lib.query.db

import io.vulpine.lib.query.util.basic.BasicPreparedListReadQuery
import mb.lib.query.model.BlastJobLink
import org.veupathdb.lib.hash_id.HashID
import java.sql.Connection
import java.sql.PreparedStatement

data class SelectParentJobLinks(private val con: Connection, private val childJobID: HashID) {
  companion object {
    private const val Query = """
    SELECT
      *
    FROM
      multiblast.multiblast_job_to_jobs
    WHERE
      job_digest = ?
    ORDER BY
      position ASC
    """
  }

  fun run(): List<BlastJobLink> = BasicPreparedListReadQuery(Query, con, ::parseJobLink, this::prep).execute().value

  private fun prep(ps: PreparedStatement) = ps.setBytes(1, childJobID.bytes)
}
